horovod/common/operations.cc

```diff
@@ -1235,7 +1266,9 @@ bool RunLoopOnce(HorovodGlobalState& state, MPIContext& ctx, bool is_coordinator
     if (state.perform_stall_check &&
         std::chrono::steady_clock::now() - state.last_stall_check >
             STALL_WARNING_TIME) {
-      CheckForStalledTensors(state, ctx);
+      // I would prefer to do "should_shut_down |= CheckForStalledTensors(state, ctx);"
+      // but for some reason it leaves the program hanging instead of quitting
```

Comment:
Can we move this before `response_list.set_shutdown(should_shut_down);` above?  Then the original `should_shut_down |= CheckForStalledTensors(state, ctx);` will work w/o getting stuck.

Reply:
done